src/main/java/org/candlepin/subscriptions/db/model/StringValueEnum.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.subscriptions.db.model;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Interface to reduce duplication of common methods for some enums
+ */
+public interface StringValueEnum {
```

Comment:
A bit of refactoring around the generics in this class would make it a bit more readable IMO (see following suggestions).
```suggestion
public interface StringValueEnum<T> {
```

Reply:
Thanks for this!  I was trying all kinds of ways to make this more readable; my generics syntax was a little rusty :smile: